posthog/api/person.py

```diff
@@ -320,17 +324,20 @@ def calculate_path_persons(self, request: request.Request) -> Dict[str, Tuple[li
 
     @action(methods=["GET"], detail=False)
     def values(self, request: request.Request, **kwargs) -> response.Response:
-        people = self.get_queryset()
-        key = "properties__{}".format(request.GET.get("key"))
-        people = people.values(key).annotate(count=Count("id")).filter(**{f"{key}__isnull": False}).order_by("-count")
-
-        if request.GET.get("value"):
-            people = people.extra(
-                where=["properties ->> %s LIKE %s"], params=[request.GET["key"], "%{}%".format(request.GET["value"])],
-            )
-
+        key = request.GET.get("key")
+        value = request.GET.get("value")
+        team = self.team
+        flattened = []
+        if key:
+            result = get_person_property_values_for_key(key, team, value)
```

Comment:
1. should we add a try/catch?
2. can we add some telemetry to this call? (metric for success/failure calls, timer for success/failure calls)

Reply:
Thanks for the review! 

1. Genuinely curious what would this add? I think it's better to bubble any exception up the stack, eventually reaching sentry, and to let the frontend handle non-200 responses.
2. I'd be happy to add telemetry, but the code I adapted doesn't come with any. What's the latest canonical way to do this?

Going on a bit of a tangent, but...

Whenever I'm in part of a codebase I don't regularly interact with, I try to copy prior art. Here I was just adapting what the equivalent events api does (without the `"custom_event"` nonsense):

https://github.com/PostHog/posthog/blob/9bb9025a90aa8312fdcc6700cd90ab3b2b6ca25e/posthog/api/event.py#L220-L236

If there are any guidelines I should follow, other than what's in the very early [coding conventions](https://posthog.com/docs/contribute/coding-conventions) doc, I don't know of them... and it would be great to add them. The small teams offsites might be a good time to collectively think of this...

Reply:
> Genuinely curious what would this add? I think it's better to bubble any exception up the stack, eventually reaching sentry, and to let the frontend handle non-200 responses.

That's a fair question! We can definitely leave everything as it is and let Sentry and the frontend handle the non-200 response. My suggestion was to explicitly handle this try/except here so that we can customise the behaviour and do like:
 
- add a metric on query/timing failure
- eventually wrap the error message body with something custom
- etc... 

> I'd be happy to add telemetry, but the code I adapted doesn't come with any. What's the latest canonical way to do this?

I don't think we have a canonical way (yet) but here's an example of it https://github.com/PostHog/posthog/blob/0464a51d7d0733cc5ebb35406884be1abb81fae4/posthog/tasks/update_cache.py#L52-L81

> If there are any guidelines I should follow, other than what's in the very early [coding conventions](https://posthog.com/docs/contribute/coding-conventions) doc, I don't know of them... and it would be great to add them. The small teams offsites might be a good time to collectively think of this...

Unfortunately there aren't yet. 100% to follow up and discuss about those collectively at the offsite!